### PR TITLE
Add test demo for database write prohibition plugin

### LINKED ISSUE #1461: Add the database write prohibition integration test
### What would you like to be enhanced?

1、Add the database write prohibition integration test.

### Why is this needed?

1、Prevent function exceptions caused by subsequent plugin code changes.

### PR DESCRIPTION
**Pull request by hanbingleixue**
**What type of PR is this?**

Integration Test

**What this PR does / why we need it?**

Add test demo for database write prohibition plugin 

**Which issue(s) this PR fixes？**

Fixes https://github.com/huaweicloud/Sermant/issues/1461

**Does this PR introduce a user-facing change?**

No

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/huaweicloud/Sermant/issues) related with this PR before you start working on it.
- [x] Make sure you have squashed your change to one single commit.
- [x] GitHub Actions works fine in this PR.

### CODE REVIEW

File: .github/actions/scenarios/database-write-prohibition/opengauss/action.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      shell: bash
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y apt-transport-https ca-certificates curl gnupg-agent software-properties-common
+        curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+        sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+        sudo apt-get update
+        sudo apt-get install -y docker-ce docker-ce-cli containerd.io
+        docker run --name opengauss -d --privileged=true  -e GS_PASSWORD=Enmo@123 -p 5432:5432 enmotech/opengauss:3.0.0
+        sudo docker ps -a
```
Comment (daizhenyu) [inline .github/actions/scenarios/database-write-prohibition/opengauss/action.yml]: need this command？

File: .github/actions/scenarios/database-write-prohibition/opengauss/action.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    - name: package demos
+      shell: bash
+      run: |
+        mvn package -Ddatabase.version=${{ matrix.opengaussVersion }} -Ddatabase.groupId=org.opengauss -Ddatabase.artifactId=opengauss-jdbc \
+        -DskipTests -Ppostgresql-test --file sermant-integration-tests/database-write-prohibition-test/pom.xml
+    - name: start opengaussv3.0.x demo
+      if: matrix.opengaussVersion == '3.0.0' || matrix.opengaussVersion == '3.0.5-og'  || matrix.opengaussVersion == '3.1.0-og'  || matrix.opengaussVersion =='3.1.1-og'
+      shell: bash
+      env:
+        SPRING_DATASOURCE_DRIVER-CLASS-NAME: org.opengauss.Driver
```
Comment (daizhenyu) [inline .github/actions/scenarios/database-write-prohibition/opengauss/action.yml]: _ and - should keep consistent

File: .github/actions/scenarios/database-write-prohibition/opengauss/action.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        SPRING_DATASOURCE_USERNAME: gaussdb
+        SPRING_DATASOURCE_PASSWORD: Enmo@123
+      run: |
+        nohup java -javaagent:sermant-agent-${{ env.sermantVersion }}/agent/sermant-agent.jar -jar \
+        -Dsermant_log_dir=${{ env.logDir }}/opengauss \
+        sermant-integration-tests/database-write-prohibition-test/postgresql-demo/target/postgresql-demo.jar > ${{ env.logDir }}/opengauss.log 2>&1 &
+    - name: waiting for services start
+      shell: bash
+      run: |
+        ps -ef | grep java
```
Comment (daizhenyu) [inline .github/actions/scenarios/database-write-prohibition/opengauss/action.yml:54]: This command can be removed.

File: .github/actions/scenarios/database-write-prohibition/opengauss/action.yml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y apt-transport-https ca-certificates curl gnupg-agent software-properties-common
+        curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+        sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+        sudo apt-get update
+        sudo apt-get install -y docker-ce docker-ce-cli containerd.io
+        docker run --name opengauss -d --privileged=true  -e GS_PASSWORD=Enmo@123 -p 5432:5432 enmotech/opengauss:3.0.0
+        sudo docker ps -a
+        ls -l && sleep 20s
```
Comment (daizhenyu) [inline .github/actions/scenarios/database-write-prohibition/opengauss/action.yml]: directly sleep

File: .github/actions/scenarios/database-write-prohibition/opengauss/action.yml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        mvn package -Ddatabase.version=${{ matrix.opengaussVersion }} -Ddatabase.groupId=org.opengauss -Ddatabase.artifactId=opengauss-jdbc \
+        -DskipTests -Ppostgresql-test --file sermant-integration-tests/database-write-prohibition-test/pom.xml
+    - name: start opengaussv3.0.x demo
+      if: matrix.opengaussVersion == '3.0.0' || matrix.opengaussVersion == '3.0.5-og'  || matrix.opengaussVersion == '3.1.0-og'  || matrix.opengaussVersion =='3.1.1-og'
+      shell: bash
+      env:
+        SPRING_DATASOURCE_DRIVER-CLASS-NAME: org.opengauss.Driver
+        SPRING_DATASOURCE_URL: jdbc:opengauss://127.0.0.1:5432/postgres?useUnicode=true&useSSL=false&characterEncoding=utf8&serverTimezone=GMT%2B8&allowPublicKeyRetrieval=true
+        SPRING_DATASOURCE_USERNAME: gaussdb
+        SPRING_DATASOURCE_PASSWORD: Enmo@123
```
Comment (daizhenyu) [inline .github/actions/scenarios/database-write-prohibition/opengauss/action.yml]: use Using a ciphertext password

File: .github/actions/scenarios/database-write-prohibition/postgresql/action.yml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        sudo -u postgres psql -c "SELECT pg_reload_conf();set password_encryption = 'md5';show password_encryption;ALTER USER postgres WITH PASSWORD '${{ env.DATASOURCE_PASSWORD }}';"
+    - name: package demos
+      shell: bash
+      run: |
+        mvn package -Ddatabase.version=${{ matrix.postgresqlVersion }} -Ddatabase.groupId=org.postgresql -Ddatabase.artifactId=postgresql \
+        -DskipTests -Ppostgresql-test --file sermant-integration-tests/database-write-prohibition-test/pom.xml
+    - name: start postgresql demo
+      shell: bash
+      env:
+        SPRING_DATASOURCE_DRIVER-CLASS-NAME: org.postgresql.Driver
```
Comment (daizhenyu) [inline .github/actions/scenarios/database-write-prohibition/postgresql/action.yml]: _ and - should keep consistent

File: .github/workflows/database_write_prohibition_integration_test.yml [hunk 7]
```diff
@@ -54,4 +55,135 @@ jobs:
...
+          - postgresqlVersion: "42.5.3"
+          - postgresqlVersion: "42.5.4"
+          - postgresqlVersion: "42.5.5"
+          - postgresqlVersion: "42.5.6"
+          - postgresqlVersion: "42.6.0"
+          - postgresqlVersion: "42.6.1"
+          - postgresqlVersion: "42.6.2"
+          - postgresqlVersion: "42.7.0"
+          - postgresqlVersion: "42.7.1"
+          - postgresqlVersion: "42.7.2"
```
Comment (daizhenyu) [inline .github/workflows/database_write_prohibition_integration_test.yml:167]: need to test postgresql all version？

File: sermant-integration-tests/database-write-prohibition-test/database-write-prohibition-integration-test/src/test/java/com/huaweicloud/sermant/database/prohibition/integration/postgresql/BatchSqlExecuteTest.java [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+    private static final String DELETE_INDEX_URL = "http://127.0.0.1:8081/postgresql/batch/deleteIndexByBatch?";
+
+    private static final String ALTER_TABLE_URL = "http://127.0.0.1:8081/postgresql/batch/alterTable?";
+
+    /**
+     * test postgresql database write prohibition function
+     */
+    @Test
+    @EnabledIfSystemProperty(named = "database.prohibition.integration.test.type", matches = "postgresql")
```
Comment (daizhenyu) [inline sermant-integration-tests/database-write-prohibition-test/database-write-prohibition-integration-test/src/test/java/com/huaweicloud/sermant/database/prohibition/integration/postgresql/BatchSqlExecuteTest.java]: postgresql use capital letter

File: sermant-integration-tests/database-write-prohibition-test/pom.xml [hunk 9]
```diff
@@ -35,12 +35,6 @@
     </dependencyManagement>
 
     <profiles>
-        <profile>
-            <id>opengauss-test</id>
-            <modules>
-                <module>opengauss-demo</module>
-            </modules>
-        </profile>
```
Comment (daizhenyu) [inline sermant-integration-tests/database-write-prohibition-test/pom.xml:43]: Keep this profile

File: .github/actions/scenarios/database-write-prohibition/opengauss/action.yml (hunk 1 above)
Comment (hanbingleixue) [reply .github/actions/scenarios/database-write-prohibition/opengauss/action.yml]: resolved

File: sermant-integration-tests/database-write-prohibition-test/pom.xml (hunk 9 above)
Comment (hanbingleixue) [reply sermant-integration-tests/database-write-prohibition-test/pom.xml:43]: The OpenGauss and PostgreSQL share a demo, which does not need to be reserved.

File: .github/actions/scenarios/database-write-prohibition/opengauss/action.yml (hunk 2 above)
Comment (hanbingleixue) [reply .github/actions/scenarios/database-write-prohibition/opengauss/action.yml]: resolved

File: .github/actions/scenarios/database-write-prohibition/opengauss/action.yml (hunk 4 above)
Comment (hanbingleixue) [reply .github/actions/scenarios/database-write-prohibition/opengauss/action.yml]: resolved

File: .github/actions/scenarios/database-write-prohibition/opengauss/action.yml (hunk 5 above)
Comment (hanbingleixue) [reply .github/actions/scenarios/database-write-prohibition/opengauss/action.yml]: resolved

File: .github/actions/scenarios/database-write-prohibition/postgresql/action.yml (hunk 6 above)
Comment (hanbingleixue) [reply .github/actions/scenarios/database-write-prohibition/postgresql/action.yml]: resolved

File: sermant-integration-tests/database-write-prohibition-test/database-write-prohibition-integration-test/src/test/java/com/huaweicloud/sermant/database/prohibition/integration/postgresql/BatchSqlExecuteTest.java (hunk 8 above)
Comment (hanbingleixue) [reply sermant-integration-tests/database-write-prohibition-test/database-write-prohibition-integration-test/src/test/java/com/huaweicloud/sermant/database/prohibition/integration/postgresql/BatchSqlExecuteTest.java]: resolved

File: .github/workflows/database_write_prohibition_integration_test.yml (hunk 7 above)
Comment (hanbingleixue) [reply .github/workflows/database_write_prohibition_integration_test.yml:167]: resolved

File: sermant-integration-tests/database-write-prohibition-test/database-write-prohibition-integration-test/src/test/java/com/huaweicloud/sermant/database/prohibition/integration/constant/DatabaseConstant.java [hunk 10]
```diff
@@ -33,6 +33,11 @@ public class DatabaseConstant {
...
     public static final String OPERATION_SUCCEED_CODE = "101";
 
+    /**
+     * successful status code
+     */
+    public static final String SUCCESS_CODE = "0";
+
     /**
      * database table data count
      */
```
Comment (lilai23) [inline sermant-integration-tests/database-write-prohibition-test/database-write-prohibition-integration-test/src/test/java/com/huaweicloud/sermant/database/prohibition/integration/constant/DatabaseConstant.java]: there exists OPERATION_SUCCEED_CODE ，no need to use another

Comment (hanbingleixue) [reply sermant-integration-tests/database-write-prohibition-test/database-write-prohibition-integration-test/src/test/java/com/huaweicloud/sermant/database/prohibition/integration/constant/DatabaseConstant.java]: 已修改

**Review by lilai23 — APPROVED** [review]

**Review by daizhenyu — APPROVED** [review]

**Pull request merged**